javalib/src/main/scala/java/lang/Integer.scala

```diff
@@ -61,6 +62,38 @@ object Integer {
   final val SIZE = 32
   final val BYTES = 4
 
+  private final val SignBit = Int.MinValue
+
+  /** Quantities in this class are interpreted as unsigned values.
+   *
+   *  - `maxLength`: `toUnsignedString(UInt.MaxValue, radix).length()`.
+   *    This is the maximum length of a string that can be parsed into an unsigned
+   *    `Int` using the given radix, assuming there is no leading sign or 0 digit.
+   *  - `overflowBarrier`: `divideUnsigned(UInt.MaxValue, radix)`.
+   *    This is the largest unsigned `x: Int` such that `x * radix` does not overflow.
+   */
+  private final class StringRadixInfo(val maxLength: Int, val overflowBarrier: Int)
+
+  /** Precomputed table for parseIntInternal. */
+  private lazy val StringRadixInfos: Array[StringRadixInfo] = {
+    val r = new Array[StringRadixInfo](Character.MAX_RADIX + 1)
+
+    for (radix <- Character.MIN_RADIX to Character.MAX_RADIX) {
+      val overflowBarrier = divideUnsigned(-1, radix)
+      var radixPower = 1
```

Comment:
```suggestion
      var radixPower = 1 // radix^(maxLength - 1)
```

The "off-by-one" nature of the loop below threw me off at first.